aspnetcore/security/authentication/cookie.md

```diff
@@ -14,335 +14,369 @@ uid: security/authentication/cookie
 
 ```csharp
-Task ValidateAsync(CookieValidatePrincipalContext context);
+var claims = new List<Claim>
+{
+    new Claim(ClaimTypes.Name, user.Email),
+    new Claim("LastUpdated"), {Database Value})
+};
+
+var claimsIdentity = new ClaimsIdentity(
+    claims, 
+    CookieAuthenticationDefaults.AuthenticationScheme);
+
+await HttpContext.SignInAsync(
+    CookieAuthenticationDefaults.AuthenticationScheme, 
+    new ClaimsPrincipal(claimsIdentity));
+```
+
+To implement an override for the `ValidateAsync` event, write a method with the following signature:
+
+```csharp
+ValidateAsync(CookieValidatePrincipalContext)
 ```
 
 ASP.NET Core Identity implements this check as part of its `SecurityStampValidator`. An example looks like the following:
 
 # [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-public static class LastChangedValidator
+public static class LastUpdatedValidator
 {
     public static async Task ValidateAsync(CookieValidatePrincipalContext context)
     {
         // Pull database from registered DI services.
-        var userRepository = context.HttpContext.RequestServices.GetRequiredService<IUserRepository>();
+        var userRepository = 
+            context.HttpContext.RequestServices
+                .GetRequiredService<IUserRepository>();
         var userPrincipal = context.Principal;
 
         // Look for the last changed claim.
-        string lastChanged;
-        lastChanged = (from c in userPrincipal.Claims
-                        where c.Type == "LastUpdated"
-                        select c.Value).FirstOrDefault();
+        var lastUpdated = (from c in userPrincipal.Claims
+                           where c.Type == "LastUpdated"
+                           select c.Value).FirstOrDefault();
 
-        if (string.IsNullOrEmpty(lastChanged) ||
-            !userRepository.ValidateLastChanged(userPrincipal, lastChanged))
+        if (string.IsNullOrEmpty(lastUpdated) ||
+            !userRepository.ValidateLastUpdated(userPrincipal, lastUpdated))
         {
             context.RejectPrincipal();
-            await context.HttpContext.SignOutAsync("MyCookieAuthenticationScheme");
+            await context.HttpContext.SignOutAsync(
+                CookieAuthenticationDefaults.AuthenticationScheme);
         }
     }
 }
 ```
 
-This would then be wired up during cookie service registration in the `ConfigureServices` method of *Startup.cs*:
+Register the event during cookie service registration in the `ConfigureServices` method:
 
 ```csharp
-services.AddAuthentication("MyCookieAuthenticationScheme")
-        .AddCookie("MyCookieAuthenticationScheme", options =>
+services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+    .AddCookie(options =>
+    {
+        options.Events = new CookieAuthenticationEvents
```

Comment:
It's worth noting that you can now use DI with the events classes in 2.0. Maybe that would a good opportunity to show that with the validator thingy.

Reply:
I agree. I'll update this later today. Thanks for the tip. Constructor injection makes a lot more sense than `GetRequiredService` ... ei ei ei *yuck!*